src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs

```diff
@@ -85,6 +87,35 @@ public HttpClient CreateClient()
 #endif
             }
 
+            string sslCaInfo = _settings.SslCaInfo;
+            if (!string.IsNullOrWhiteSpace(sslCaInfo))
```

Comment:
Just above this we check if `GIT_SSL_NO_VERIFY` (or the git config equivalent) has been set and set the certificate validation callback(s) to return `true`.

If both `GIT_SSL_NO_VERIFY` **_and_** `http.sslCAInfo` have been set, what should the outcome be (which callback is run) if the root certificate doesn't match? Does "no verify" win? We should match Git's behaviour.

Should probably join these blocks together in an `if-elseif`, and also validate _here_ if we should use `sslCAInfo` based on `sslBackend` (not in `ISettings`).

Reply:
According to the Git implementation and [documentation](https://curl.se/libcurl/c/CURLOPT_CAINFO.html) of the cURL parameters it uses, `GIT_SSL_NO_VERIFY`/`http.sslVerify` should take precedence. I'll update the blocks to use an `if-elseif` structure accordingly.